From: Alec Leamas <leamas@nowhere.net>
Date: Mon, 30 Jul 2018 13:18:12 -0400
Subject: CMakelists: Fix FTBS on Debian sid (+ likely other gtk3 platforms)

On platforms using gtk3, invoke  FIND_PACKAGE(wxWidgets) with proper
options to handle those who have mgtk2 as default, like Debian sid.
Also add the __WXGTK3__ symbol which seems to be missing from the
wx-config output on Debian.

Here is plenty of room for clean-ups; the condition WIN32 or
APPLE or QT_ANDROID comes to mind.
---
 CMakeLists.txt | 20 +++++++++++++-------
 1 file changed, 13 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b702c42..4ed9f9f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -440,14 +440,16 @@ set (WXWIDGETS_FORCE_VERSION CACHE VERSION "Force usage of a specific wxWidgets
 if(WXWIDGETS_FORCE_VERSION)
   set (wxWidgets_CONFIG_OPTIONS --version=${WXWIDGETS_FORCE_VERSION})
 endif()
-FIND_PACKAGE(wxWidgets REQUIRED)
 
-IF(MSYS)
-# this is to convert msys to windows paths, and handle the missing /usr
-STRING( REGEX REPLACE "/usr/local" ";C:/MinGW/msys/1.0/local" wxWidgets_INCLUDE_DIRS ${wxWidgets_INCLUDE_DIRS} )
-ENDIF(MSYS)
+IF(WIN32 OR APPLE OR QT_ANDROID)
+  FIND_PACKAGE(wxWidgets REQUIRED)
+  IF(MSYS)
+  # this is to convert msys to windows paths, and handle the missing /usr
+  STRING( REGEX REPLACE "/usr/local" ";C:/MinGW/msys/1.0/local" wxWidgets_INCLUDE_DIRS ${wxWidgets_INCLUDE_DIRS} )
+  ENDIF(MSYS)
+  INCLUDE(${wxWidgets_USE_FILE})
+ENDIF(WIN32 OR APPLE OR QT_ANDROID)
 
-INCLUDE(${wxWidgets_USE_FILE})
 MESSAGE (STATUS "Found wxWidgets..." )
     MESSAGE (STATUS " wxWidgets Includ: ${wxWidgets_INCLUDE_DIRS}")
     MESSAGE (STATUS " wxWidgets Libraries: ${wxWidgets_LIBRARIES}")
@@ -910,6 +912,7 @@ IF(NOT WIN32 AND NOT APPLE AND NOT QT_ANDROID)
   ENDIF(NOT OCPN_FORCE_GTK3)
 
   IF(GTK2_FOUND)
+    set(wxWidgets_CONFIG_OPTIONS ${wxWidgets_CONFIG_OPTIONS} --toolkit=gtk2)
     INCLUDE_DIRECTORIES(${GTK2_INCLUDE_DIRS})
     SET(GTK_LIBRARIES ${GTK2_LIBRARIES})
     MESSAGE(STATUS "Building against GTK2...")
@@ -917,9 +920,12 @@ IF(NOT WIN32 AND NOT APPLE AND NOT QT_ANDROID)
     FIND_PACKAGE(GTK3)
     INCLUDE_DIRECTORIES(${GTK3_INCLUDE_DIRS})
     SET(GTK_LIBRARIES ${GTK3_LIBRARIES})
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__WXGTK3__")
+    set(wxWidgets_CONFIG_OPTIONS ${wxWidgets_CONFIG_OPTIONS} --toolkit=gtk3)
     MESSAGE(STATUS "Building against GTK3...")
   ENDIF(GTK2_FOUND)
-        
+  FIND_PACKAGE(wxWidgets REQUIRED)
+  INCLUDE(${wxWidgets_USE_FILE})
   SET(EXTRA_LIBS ${EXTRA_LIBS} ${GTK_LIBRARIES})
 ENDIF(NOT WIN32 AND NOT APPLE AND NOT QT_ANDROID)
 
